system/security/plugins/jwtSecurity/jwtSecurity.cpp

```diff
@@ -795,9 +802,9 @@ class CJwtSecurityManager : implements IDaliLdapConnection, public CBaseSecurity
                         // Scope hpccinternal::<username> always has full access for their own scope, but
                         // explicitly denied when attempting to access someone else's
                         // hpccinternal::<username> scope
-                        if (resourceName && strncmp(resourceName, "hpccinternal::", 14) == 0)
+                        if (strisame(resourceName, hpccInternalScope.c_str()))
                         {
-                            if (strisame(&resourceName[14], user.getName()))
+                            if (strisame(&resourceName[hpccInternalScope.length()], user.getName()))
```

Comment:
Same comment as above, note how LDAPSecurityManager does it by extracting the name from the filescope

        StringBuffer userName;
        for (const char * p = &filescope[m_hpccInternalScope.length()]; *p && *p != ':'; p++)//extract scope username
            userName.append(*p);
        if(strieq(userName.str(), user.getName()))
            return SecAccess_Full;
        PROGLOG("Access denied to scope %s for user %s", filescope, user.getName());
        return SecAccess_None;

Reply:
which begs the question, how did it work (or did it?) before ? unless the resourceName string always terminates after the username??

Reply:
The resource name did, in fact, terminate immediately after the username.  IIRC from testing the original code, I never saw any "deeper" file scope passed in here.  That said, I will revisit and retest once I debug my test setup.

Reply:
I need to find whatever drugs I was on back then and make sure I never take them again.

Testing this morning showed that Russ was right and the code was wrong.  It is difficult to see how it ever worked.

I've made the changes Russ suggested and have verified that big jobs that use hpccinternal scopes work.